Vban fixes

pblivingston opened the pull request:
Closes #18 

## What

Adds Vban.port and implements a handful of fixes and small changes to Vban

### Added

- Vban.port sets Vban.Instream[0].port
- Midi and Command streams now support write-only:
  - on
  - name
  - ip

### Changed

- some vban.instream | vban.outstream commands now added with meta functions
  - on
  - name
  - ip
- cast vban getters to types for consistency

### Fixed

- some vban commands incorrectly read-only/write-only
  - enable
  - instream|outstream.quality
  - instream|outstream.route
- vban.stream.port: [string]$arg -> [int]$arg
- vban route range (API documentation is incorrect)
- vban.stream.sr: $this._port -> $this._sr

## Test summary

Pester tests pass for all kinds, no manual tests needed

pblivingston committed:
Update vban.ps1

- vban.port
- on -> AddBoolMembers
- name, ip -> AddStringMembers
- cast getters to types
- correct read-only/write-only
- correct route range

pblivingston committed:
Update higher.Tests.ps1

pester tests pass for all kinds
- increased sleeps after restarts to 2s
- added tags to test types

pblivingston committed:
update docs

onyx-and-iris commented:
This is great, it's worth pointing out that in a recent update to the API its possible to run some commands for the MIDI and TEXT streams so we might consider supporting those.

One possible implementation would involve updating the kind maps and adding extra VbanInstream/VbanOutstream instances to the vban.instream/outstream arrays. Take a look at [this example gist](https://gist.github.com/onyx-and-iris/c107715611f249469698a12e8773fee2).

Let me know your thoughts.

pblivingston commented:
That looks like a good structure. I expect 'on', 'name', and 'ip' will work for all and 'route' will work for midi out. Certain others might work as read-only, but their information wouldn't be useful, so not even worth checking. Though I will confirm whether 'port' can be set simply so it doesn't need to be changed if per-stream ports are supported in the future.

I'll re-mark this ready for review once that's finished!

Your mentioning prompted me to actually look at the version history thread - looks like there are a few more things I wasn't aware of that I can add to my to-do list for future PRs!

pblivingston committed:
midi/text streams

- added midi and text streams with:
  - on
  - name
  - ip
had to manually test as these are currently write-only, but I suspect this is a bug/will change in the future

pblivingston committed:
Update CHANGELOG.md

pblivingston commented:
Added the extra streams with On, Name, and IP. The commands are currently write-only, but I suspect this is a bug or will be changed in the future. If that gets fixed we can simply move
```powershell
AddBoolMembers -PARAMS @('on')
AddStringMembers -PARAMS @('name', 'ip')
```
from VbanStream to Vban, delete the write-only commands, and uncomment the indexes in higher.Tests.ps1.

pblivingston committed:
'stream' -> 'audio'

VbanStream -> VbanAudio for clarity

onyx-and-iris commented:
Alright, very nice I like what you've done here. That's a strange bug, probably worth reporting.

pblivingston commented:
Looks like I registered for a forum account back in 2020 but it was never approved. I sent them an email to see if I can get that activated and open a topic for the bug there.

... except it seems like my IP is now banned or something. I can't access the forums from my home network.

onyx-and-iris commented:
The Voicemeeter forum requires manual activation, it's a bit annoying tbh... the discord server is probably a better place to report it. Don't worry I'll mention it.

pblivingston commented:
Okay great, thanks!
